### PR TITLE
Create MD (markdown) file that explains the workflow

### PR DESCRIPTION
**Pull request by ggrig**

### CODE REVIEW

**Comment by yardenshoham** [comment]
Closes #11

File: docs/rebuild-engine-as-a-target.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## Basic steps
+
+1. A file to be processed is added to the file queue
+2. The Inspector POD gets the "File Added" event
+3. The Inspector POD lunches a File Processing POD
+4. The File Processing POD gets the file from the queue 
+5. The File Processing POD rebuilds the file with GW Rebuild API
+6. The File Processing POD logs the rebuild results/errors to the Logging POD
+7. The Inspector POD destroys the File Processing POD as a potentially dangerous one 
+
```
Comment (DinisCruz) [inline docs/rebuild-engine-as-a-target.md:20]: This workflow description is spot on 👍

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+There will be two types of scenarios: random and sequential. 
+
+Once a random scenario is read, the traffic generator picks up instructions from the scenario randomly and executing them. This means that some of the instructions will be executed multiple times. To increase the probability that all the instructions from a random scenario are executed, the traffic generator will run significantly more iterations than the number of instructions in the scenario. 
+
+In the case of a sequential scenario, the traffic generator will execute all the instructions in the order they are placed in the scenario file until all the instructions are executed. 
+
+Every POD can contain multiple scenario files. The traffic generator will pick up a single scenario at a time, execute it, and then move to the next (randomly picked up) scenario. The process will repeat in an infinite loop.
+
+![traffic](pngs/traffic-generator.png)
+
```
Comment (DinisCruz) [inline docs/README.md:16]: On this mode, replace URL with Action

Those actions will be specific to a target website

I like the idea to either define the Actions sequence of events , or to have it running at random

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+## Multiple sites support
+
+To support customized actions for different URLs on different websites, the traffic generators will implement expandable APIs for supported actions (open a page, follow a link, upload a file, download a file).
+
+Support of different web-site will be modularized into plugins. This will allow one to easily add/remove the support of necessary/unnecessary websites on a particular POD.  
+
+When executing a particular instruction the API will check if any of the plugins support the particular instruction. If yes, the handling of the instruction will be passed to the plugin. Otherwise, generic processing will be applied. 
+
+![multiple-sites](pngs/traffic-for-multiple-sites-API-v2.png)
+
```
Comment (DinisCruz) [inline docs/README.md:26]: Please drop this mode, you are going on an ‘over-engineering’ path

We don’t this this type of modularised module

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+Traffic generation PODs will be allocated and deleted dynamically. So storing any kind of info withing the PODs won't make much sense. Because of that logs (traces, metrics) are to be transferred to a persistent location. Special PODs running log/metrics collection/analysis tools will be utilized for the purpose. 
+
+The following options are currently considered to be utilized as logs/traces/metrics monitoring tools: 
+- https://www.elastic.co/
+- https://www.jaegertracing.io/
+- https://min.io/
+- http://logio.org/
+- https://opencensus.io/
+
+![logging](pngs/logging-workflow.png)
```
Comment (DinisCruz) [inline docs/README.md:39]: Yes 👍

**Review by DinisCruz — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 3 above)
Comment (ggrig) [reply docs/README.md:26]: Hi Dinis,
Do you mean that no POD will need supporting multiple sites?
As a consequence, do you say we do not need to have plugins there?

File: docs/README.md (hunk 2 above)
Comment (ggrig) [reply docs/README.md:16]: Done

File: docs/README.md (hunk 3 above)
Comment (ggrig) [reply docs/README.md]: Done

**Review by DinisCruz — APPROVED** [review]

**Pull request merged**